test/jit/test_tracer.py

```diff
@@ -2367,3 +2367,41 @@ def forward(
         with self.assertRaisesRegex(RuntimeError, "cannot be understood by the tracer, only outputs matching"):
             mod = ReturnsBadDict()
             traced_module = torch.jit.trace(mod, [torch.ones(1), torch.ones(1)], strict=False)
+
+    def test_trace_func_argument_names_captured(self):
```

Comment:
Since the motivation for the PR includes

> This is useful for both debugging and supporting using InterfaceType to represent traced modules.

consider adding a test to check that a traced module can indeed be used as the implementation of an interface type.

Reply:
Done. Also moved test cases to proper parent test class